spec/lib/gateway/stat_gateway_spec.rb

```diff
@@ -0,0 +1,126 @@
+  end
+
+  context 'given no signups' do
+    it 'returns zero signups for today' do
+      expect(subject.signups[:today]).to eq(0)
+    end
+
+    it 'returns zero signups for today' do
+      expect(subject.signups[:total]).to eq(0)
+    end
+  end
+
+  context 'given 3 signups today' do
+    before do
+      3.times do |i|
+        User.create(username: "Today #{i}")
+      end
+    end
+
+    it 'returns signups for today' do
+      expect(subject.signups[:today]).to eq(3)
+    end
+
+    it 'returns same total number of signups' do
+      expect(subject.signups[:total]).to eq(3)
+    end
+  end
+
+  context 'given 5 signups today and 1 yesterday' do
+    before do
+      User.create(username: 'Yesterday', created_at: Date.today - 1)
+
+      5.times do |i|
+        User.create(username: "Today #{i}")
+      end
+    end
+
+    it 'returns zero signups 5 signups for today' do
+      expect(subject.signups[:today]).to eq(5)
+    end
+
+    it 'returns zero signups 6 signups total' do
+      expect(subject.signups[:total]).to eq(6)
+    end
+  end
+
+  context 'given 2 signups tomorrow' do
+    before do
+      2.times do |i|
+        User.create(username: "Tomorrow #{i}", created_at: Date.today + 1)
+      end
+    end
+
+    it 'returns zero signups for today' do
+      expect(subject.signups[:today]).to eq(0)
+    end
+
+    it 'returns zero signups total' do
+      expect(subject.signups[:today]).to eq(0)
+    end
+  end
+
+  context 'given 1 SMS signup today' do
+    before do
+      User.create(username: "non-SMS")
+
+      User.create(
+        username: "SMS", 
+        contact: '+0123456789',
+        sponsor: '+0123456789'
+        )
+    end
+
+    it 'counds them against total singups' do
+      expect(subject.signups[:total]).to eq(2)
+    end
+
+    it 'counds them against today\'s signups' do
+      expect(subject.signups[:today]).to eq(2)
+    end
+
+    it 'counds them against SMS total signups' do
+      expect(subject.signups[:sms_total]).to eq(1)
+    end
+
+    it 'counds them against SMS today\'s signups' do
+      expect(subject.signups[:sms_today]).to eq(1)
+    end
+  end
+
+  context 'given SNS signups made on different dates' do
+    before do
+      User.create(
+        username: "SMS old", 
+        created_at: Date.today -  1,
+        contact: '+1123456789',
+        sponsor: '+1123456789'
+        )
+
+      User.create(
+        username: "SMS today", 
+        contact: '+0123456789',
+        sponsor: '+0123456789'
+        )
+    end
+
+    it 'counts them against total singups' do
+      expect(subject.signups[:total]).to eq(2)
+    end
+
+    it 'counds them against today\'s signups' do
+      expect(subject.signups[:today]).to eq(1)
+    end
+
+    it 'counds them against SMS total signups' do
+      expect(subject.signups[:sms_total]).to eq(2)
+    end
+
+    it 'counds them against SMS today\'s signups' do
+      expect(subject.signups[:sms_today]).to eq(1)
+    end
+  end
+end
```

Comment:
Some of those tests will quickly become redundant and noisy once we add two last stats. I don't want to create a mega PR so I ought to merge it now and refactor once we have the whole gateway working.

Reply:
Sounds like a plan.